src/librustc_mir/dataflow/framework/backward.rs

```diff
@@ -0,0 +1,172 @@
+use rustc_index::bit_set::BitSet;
+use rustc_middle::mir::{self, BasicBlock, Location};
+
+use super::{Analysis, Effect, EffectIndex, GenKillAnalysis, GenKillSet};
+
+/// Applies all effects between locations `from` and `to` inclusive.
+pub fn apply_effects_in_range_inclusive<A>(
+    analysis: &A,
+    state: &mut BitSet<A::Idx>,
+    block: BasicBlock,
+    block_data: &mir::BasicBlockData<'tcx>,
+    from: EffectIndex,
+    to: EffectIndex,
```

Comment:
```suggestion
    range: RangeInclusive<EffectIndex>,
```
To take advantage of range notation?